src/components/modules/api/index.ts

```diff
@@ -17,21 +17,50 @@ export default class API extends Module {
    * Editor.js Core API modules
    */
   public get methods(): APIInterfaces {
-    return {
-      blocks: this.Editor.BlocksAPI.methods,
-      caret: this.Editor.CaretAPI.methods,
-      events: this.Editor.EventsAPI.methods,
-      listeners: this.Editor.ListenersAPI.methods,
-      notifier: this.Editor.NotifierAPI.methods,
-      sanitizer: this.Editor.SanitizerAPI.methods,
-      saver: this.Editor.SaverAPI.methods,
-      selection: this.Editor.SelectionAPI.methods,
-      styles: this.Editor.StylesAPI.classes,
-      toolbar: this.Editor.ToolbarAPI.methods,
-      inlineToolbar: this.Editor.InlineToolbarAPI.methods,
-      tooltip: this.Editor.TooltipAPI.methods,
-      i18n: this.Editor.I18nAPI.methods,
-    } as APIInterfaces;
+    const apiComponents = {
+      blocks: this.Editor.BlocksAPI,
+      caret: this.Editor.CaretAPI,
+      events: this.Editor.EventsAPI,
+      listeners: this.Editor.ListenersAPI,
+      notifier: this.Editor.NotifierAPI,
+      sanitizer: this.Editor.SanitizerAPI,
+      saver: this.Editor.SaverAPI,
+      selection: this.Editor.SelectionAPI,
+      styles: this.Editor.StylesAPI,
+      toolbar: this.Editor.ToolbarAPI,
+      inlineToolbar: this.Editor.InlineToolbarAPI,
+      tooltip: this.Editor.TooltipAPI,
+      i18n: this.Editor.I18nAPI,
+      readOnly: this.Editor.ReadOnlyAPI,
+    };
+
+    const allMethods = {} as APIInterfaces;
+
+    for (const apiComponent in apiComponents) {
```

Comment:
maybe we can place this code to the ApiModule?

Reply:
ApiModule is an abstraction for single class, so in that function I have to call decoration process in each ApiModule extension. It was easier to all for "ApiModules" on methods getter so that it is in one place